ssds/__init__.py

```diff
@@ -181,28 +175,22 @@ def _upload_oneshot(self, url: str, ssds_key: str) -> str:
         return ssds_key
 
     def _upload_multipart(self,
-                          url: str,
+                          src_blob: Blob,
                           ssds_key: str,
                           part_size: int,
                           threads: Optional[int]) -> str:
         checksums: Dict[str, Union[str, checksum.UnorderedChecksum]]
-        if url.startswith("s3://"):
-            bucket_name, key = url[5:].split("/", 1)
-            s3_blob = S3Blob(bucket_name, key)
-            parts = s3_blob.parts(threads)
-            # s3 -> s3 must recompute the s3etag because the part layout may change
+        if isinstance(src_blob, S3Blob):
```

Comment:
determining which checksums to compute requires two parameters, `src_blobstore` and `dst_blobstore`.

In order to return the required checksums, Blobs would need to be instantiated with an additional parameter, `dst_blobstore`. Since the final destination of a blob may change, or be entirely unused, I feel it is better to handle this state externally to the Blobstore abstractions.